nestkernel/connection_creator_impl.h

```diff
@@ -714,7 +711,7 @@ ConnectionCreator::fixed_outdegree_( Layer< D >& source,
       probabilities.resize( target_pos_node_id_pairs.size(), 1.0 );
     }
 
-    const auto number_of_connections = std::round( number_of_connections_->value( grng, src ) );
+    long number_of_connections = std::round( number_of_connections_->value( grng, src ) );
```

Comment:
The [new warning](https://github.com/nest/nest-simulator/actions/runs/4231025828/jobs/7349173899#step:9:657) (that is now failing the build) suggests that this should rather be a `size_t`

Reply:
The [second new warning](https://github.com/nest/nest-simulator/actions/runs/4231025828/jobs/7349173899#step:9:664) seems to point to the same problem.

Reply:
I think the signed type used there is the main issue here. So using `unsigned long` might be the way to go here. Of course, simply using `size_t` would also work and might also make sense as we are comparing to `size_t` here. Without knowing that we compare against a `size_t` (which one does in theory not know at the time of the declaration), `unsigned long` would be the type I would use here, as this would be the correct type for a simple counter. What do you think?

Reply:
I think your reasoning `unsigned long` is sound. Go for it!